tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,395 @@
+    const USER_A_EMAIL = 'user_a@test.com';
+    const USER_B_ACCOUNT_ID = 2;
+    const USER_B_EMAIL = 'user_b@test.com';
+    const USER_C_ACCOUNT_ID = 3;
+    const USER_C_EMAIL = 'user_c@test.com';
+
+    // Render the App and sign in as a test user.
+    const renderedApp = render(<App />);
+
+    // Note: act() is necessary since react-navigation's NavigationContainer has an internal state update that will throw some
+    // warnings related to async code. See: https://callstack.github.io/react-native-testing-library/docs/understanding-act/#asynchronous-act
+    await act(async () => {
+        await waitForPromisesToResolve();
+        await waitForPromisesToResolve();
+    });
+
+    const loginForm = renderedApp.getByTestId('login-form');
+    expect(loginForm).toBeTruthy();
+
+    TestHelper.signInWithTestUser(USER_A_ACCOUNT_ID, USER_A_EMAIL, undefined, undefined, 'A');
+    await waitForPromisesToResolve();
+
+    const MOMENT_TEN_MINUTES_AGO = moment().subtract(10, 'minutes');
+
+    // Simulate setting an unread report and personal details
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${REPORT_ID}`, {
+        reportID: REPORT_ID,
+        reportName: 'Chat Report',
+        maxSequenceNumber: 9,
+        lastReadSequenceNumber: 1,
+        lastMessageTimestamp: MOMENT_TEN_MINUTES_AGO.utc(), // Ten minutes ago
+        lastMessageText: 'Test',
+        participants: [USER_B_EMAIL],
+    });
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`, {
+        0: {
+            actionName: CONST.REPORT.ACTIONS.TYPE.CREATED,
+            automatic: false,
+            sequenceNumber: 0,
+            timestamp: MOMENT_TEN_MINUTES_AGO.unix(),
+            reportActionID: NumberUtils.rand64(),
+        },
+        1: TestHelper.buildTestReportComment(USER_B_EMAIL, 1, MOMENT_TEN_MINUTES_AGO.add(10, 'seconds').unix()),
+        2: TestHelper.buildTestReportComment(USER_B_EMAIL, 2, MOMENT_TEN_MINUTES_AGO.add(20, 'seconds').unix()),
+        3: TestHelper.buildTestReportComment(USER_B_EMAIL, 3, MOMENT_TEN_MINUTES_AGO.add(30, 'seconds').unix()),
+        4: TestHelper.buildTestReportComment(USER_B_EMAIL, 4, MOMENT_TEN_MINUTES_AGO.add(40, 'seconds').unix()),
+        5: TestHelper.buildTestReportComment(USER_B_EMAIL, 5, MOMENT_TEN_MINUTES_AGO.add(50, 'seconds').unix()),
+        6: TestHelper.buildTestReportComment(USER_B_EMAIL, 6, MOMENT_TEN_MINUTES_AGO.add(60, 'seconds').unix()),
+        7: TestHelper.buildTestReportComment(USER_B_EMAIL, 7, MOMENT_TEN_MINUTES_AGO.add(70, 'seconds').unix()),
+        8: TestHelper.buildTestReportComment(USER_B_EMAIL, 8, MOMENT_TEN_MINUTES_AGO.add(80, 'seconds').unix()),
+        9: TestHelper.buildTestReportComment(USER_B_EMAIL, 9, MOMENT_TEN_MINUTES_AGO.add(90, 'seconds').unix()),
+    });
+    Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, {
+        [USER_B_EMAIL]: TestHelper.buildPersonalDetails(USER_B_EMAIL, USER_B_ACCOUNT_ID, 'B'),
+    });
+    await waitForPromisesToResolve();
```

Comment:
Again here, adding `await` to each of the `merge()` methods should allow this `waitForPromisesToResolve()` to be removed.

Reply:
Ah in this case, I think we are not just waiting for `Onyx.merge()` but also some fun stuff that "eventually happens" after the merge.

Reply:
Hm, maybe this is the cause to some of my test failures that I'm seeing on my new sidebarlinks PR. I need to explore it further.